libs/@local/hashql/core/src/module/namespace.rs

```diff
@@ -0,0 +1,363 @@
+//! Namespace management for the HashQL module system.
+use super::{ModuleRegistry, import::Import, item::Universe};
+use crate::symbol::InternedSymbol;
+
+/// Represents the namespace of a module.
+///
+/// A `ModuleNamespace` defines the collection of names that are available within a module.
+#[derive(Debug, Clone)]
+pub struct ModuleNamespace<'env, 'heap> {
+    registry: &'env ModuleRegistry<'heap>,
+    pub imports: Vec<Import<'heap>>,
+}
+
+impl<'env, 'heap> ModuleNamespace<'env, 'heap> {
+    pub const fn new(registry: &'env ModuleRegistry<'heap>) -> Self {
+        Self {
+            registry,
+            imports: Vec::new(),
+        }
+    }
+
+    /// Attempts to add an absolute import.
+    ///
+    /// An absolute import directly references an item from the global module registry
+    /// using a fully-qualified path.
+    ///
+    /// # Returns
+    ///
+    /// `bool` - `true` if the import was successful, `false` if the item wasn't found.
+    pub fn import_absolute(
+        &mut self,
+        name: InternedSymbol<'heap>,
+        query: impl IntoIterator<Item = InternedSymbol<'heap>, IntoIter: Clone>,
+    ) -> bool {
+        let results = self.registry.search(query);
+        if results.is_empty() {
+            return false;
+        }
+
+        for result in results {
+            self.imports.push(Import {
+                name,
+                item: result.id,
+                universe: result.kind.universe(),
+            });
+        }
+
+        true
+    }
+
+    /// Attempts to add a relative import.
+    ///
+    /// A relative import can be resolved either:
+    /// 1. As an absolute import if it exists in the global registry
+    /// 2. Relative to another import already in scope
+    ///
+    /// # Returns
+    ///
+    /// `bool` - `true` if the import was successfully resolved, `false` if the item wasn't found.
```

Comment:
```suggestion
    /// `true` if the import was successfully resolved, `false` if the item wasn't found.
```

Reply:
fixed in [8966d6b](https://github.com/hashintel/hash/pull/7014/commits/8966d6b9c92094ec602c2fd0eeea4dadbe1930b1)